package/test/test_email_service.py

```diff
@@ -309,3 +309,38 @@ def test_is_not_email_configured(self):
         self.email_service._email_config = None
         self.assertFalse(self.email_service.is_email_configured())
 
+    @patch('smtplib.SMTP')
+    def test_validate_email_config(self, mock_smtp):
+        # arrange
+        smtp_obj = Mock()
+        smtp_obj.ehlo = Mock()
+        smtp_obj.starttls = Mock()
+        smtp_obj.login = Mock()
+        smtp_obj.close = Mock()
+        mock_smtp.return_value = smtp_obj
+
+        # act
+        self.email_service.validate_email_config()
+
+        # assert
+        mock_smtp.return_value.ehlo.assert_called_once()
+        mock_smtp.return_value.starttls.assert_called_once()
+        mock_smtp.return_value.login.assert_called_once()
+        mock_smtp.return_value.close.assert_called_once()
+
+    @patch('smtplib.SMTP', side_effect=Exception())
+    def test_validate_email_config_exception(self, mock_smtp):
+        # arrange
+        # side_effect=Exception()
+
+        # act
+        excepted = False
+        try:
+            self.email_service.validate_email_config()
+        except Exception as e:
+            excepted = True
+            # assert
+            self.assertEqual(e.args[0], 'Failed to login to SMTP server')
+        # assert
+        self.assertTrue(excepted)
```

Comment:
1. If there is an error I think we will get an exception during the "login method", so this is the method that should get the side_effect.
2. There is a better way "assert" if there is an exception, the common pattern is:
with self.assertRaises(TypeError):
    ...do something
3. It's better to catch a more specific error. Exception is the most high level error type so if we only look if we get Exception we might get false positive results and miss real bugs